framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1238,43 +1263,30 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
-        }
-        TreePath path = atypeFactory.getPath(node);
+    protected void checkSuperConstructorCall(MethodInvocationTree superCall) {
+        TreePath path = atypeFactory.getPath(superCall);
         MethodTree enclosingMethod = TreeUtils.enclosingMethod(path);
-        if (TreeUtils.isConstructor(enclosingMethod)) {
-            AnnotatedTypeMirror superType = atypeFactory.getAnnotatedType(node);
-            AnnotatedExecutableType constructorType =
-                    atypeFactory.getAnnotatedType(enclosingMethod);
-            Set<? extends AnnotationMirror> topAnnoations =
-                    atypeFactory.getQualifierHierarchy().getTopAnnotations();
-            for (AnnotationMirror topAnno : topAnnoations) {
-                AnnotationMirror superTypeMirror = superType.getAnnotationInHierarchy(topAnno);
-                AnnotationMirror constructorTypeMirror =
-                        constructorType.getReturnType().getAnnotationInHierarchy(topAnno);
-
-                if (!AnnotationUtils.areSameByName(
-                                constructorTypeMirror,
-                                "org.checkerframework.checker.initialization.qual.UnderInitialization")
-                        && !AnnotationUtils.areSameByName(
+        AnnotatedTypeMirror superType = atypeFactory.getAnnotatedType(superCall);
+        AnnotatedExecutableType constructorType = atypeFactory.getAnnotatedType(enclosingMethod);
+        Set<? extends AnnotationMirror> topAnnotations =
+                atypeFactory.getQualifierHierarchy().getTopAnnotations();
+        for (AnnotationMirror topAnno : topAnnotations) {
+            AnnotationMirror superTypeMirror = superType.getAnnotationInHierarchy(topAnno);
+            AnnotationMirror constructorTypeMirror =
+                    constructorType.getReturnType().getAnnotationInHierarchy(topAnno);
+
+            if (!atypeFactory
+                    .getQualifierHierarchy()
+                    .isSubtype(superTypeMirror, constructorTypeMirror)) {
+                checker.report(
+                        Result.failure(
+                                "super.invocation.invalid",
                                 constructorTypeMirror,
-                                "org.checkerframework.checker.nullness.qual.Raw")
-                        && !atypeFactory
-                                .getQualifierHierarchy()
-                                .isSubtype(superTypeMirror, constructorTypeMirror)) {
-                    checker.report(
-                            Result.failure(
-                                    "super.invocation.invalid",
-                                    constructorTypeMirror,
-                                    node,
-                                    superTypeMirror),
-                            enclosingMethod);
-                }
+                                superCall,
+                                superTypeMirror),
+                        enclosingMethod);
```

Comment:
Currently the error is issued on the constructor declaration. Should we use the superCall MethodInvocationTree instead?

Reply:
I did that. Thanks for pointing this out!